New updates to QPU info/get backend info pages (#3803)
Qiskit/documentation: docs/guides/qpu-information.mdx

# QPU information

> IBM&reg; offers both open and premium access to a wide variety of quantum processing units (QPUs). All QPUs deployed by IBM are based on superconducting qubit technology, as the control and scalability of this technology pave a clear path to achieving quantum advantage with these QPUs. You can see the full details of all IBM QPUs on the [Compute resources](https://quantum.cloud.ibm.com/computers) page.

Reviewer: ```suggestion
IBM&reg; offers access to a wide variety of quantum processing units (QPUs). All QPUs deployed by IBM are based on superconducting qubit technology, as the control and scalability of this technology pave a clear path to achieving quantum advantage with these QPUs.

Explore all public IBM QPUs by navigating to the [Compute resources](https://quantum.cloud.ibm.com/computers) page on the IBM Quantum&reg; Platform. Click on any QPU to open its detailed information card.

In this page, we describe the detailed information you will find on the QPU information card, as well as using [Qiskit](/docs/api/qiskit-ibm-runtime/ibm-backend).
```

*   Small electronics changes that don’t affect operation.
*   QPU software updates.

## QPU details

> Find details about a specific QPU on the [Compute resources](https://quantum.cloud.ibm.com/computers) page by selecting the QPU. The card that opens up displays the following details: 

Reviewer: ```suggestion
The first section of the QPU information card focusses on overall QPU information: 
```

*   Small electronics changes that don’t affect operation.
*   QPU software updates.

## QPU details

The first section of the QPU information card provides the following QPU details: 

* **Name** - The unique name assigned to a specific QPU. QPUs hosted on IBM Cloud® have names that begin with `ibm_*`. All QPUs are given a city name, e.g., `ibm_kingston`. This name does not indicate where the actual QPU is hosted. They are named after IBM&reg; locations around the world.
* **Qubits** - The number of physical qubits in a QPU.

> * **2Q error (layered)** - Average error per layered gate (EPLG) in a chain of 100 qubits. Average EPLG measures the average gate error in a layered chain of $N$ qubits ($N$=100 here). It is derived from a similar quantity known as the layer fidelity (LF) where EPLG$_{100}$ = 4/5(1-LF$^{\frac{1}{99}}$) and layer fidelity is the process fidelity of the layered chain of $N$ qubits. For details, see the paper [Benchmarking quantum processor performance at scale](https://arxiv.org/abs/2311.05933). Note that in the paper EPLG is defined for process error, but for consistency with the individually reported gate errors here it is quoted for average gate error, thus the factor of 4/5.

Reviewer 2: Add here a link to an example notebook https://github.com/qiskit-community/qiskit-device-benchmarking/blob/main/notebooks/layer_fidelity_single_chain.ipynb

> * **2Q error (layered)** - Average error per layered gate (EPLG) in a chain of 100 qubits. Average EPLG measures the average gate error in a layered chain of $N$ qubits ($N$=100 here). It is derived from a similar quantity known as the layer fidelity (LF) where EPLG$_{100}$ = 4/5(1-LF$^{\frac{1}{99}}$) and layer fidelity is the process fidelity of the layered chain of $N$ qubits. For details, see the paper [Benchmarking quantum processor performance at scale](https://arxiv.org/abs/2311.05933). Note that in the paper EPLG is defined for process error, but for consistency with the individually reported gate errors here it is quoted for average gate error, thus the factor of 4/5.

Author: ```suggestion
* **2Q error (layered)** - Average error per layered gate (EPLG) in a chain of 100 qubits. Average EPLG measures the average gate error in a layered chain of $N$ qubits ($N$=100 here). It is derived from a similar quantity known as the layer fidelity (LF) where EPLG$_{100}$ = 4/5(1-LF$^{\frac{1}{99}}$) and layer fidelity is the process fidelity of the layered chain of $N$ qubits. For details, see the paper [Benchmarking quantum processor performance at scale](https://arxiv.org/abs/2311.05933). Note that in the paper EPLG is defined for process error, but for consistency with the individually reported gate errors here it is quoted for average gate error, thus the factor of 4/5.  For an example experiment that runs layer fidelity on a 100-qubit chain, see [this notebook](https://github.com/qiskit-community/qiskit-device-benchmarking/blob/main/notebooks/layer_fidelity_single_chain.ipynb) from the Qiskit Community GitHub.
```

> * **CLOPS** (or CLOPS_h) - Circuit layer operations per second, is a measure of how many layers of a 100x100 circuit (hardware-aware circuit) a QPU (quantum processing unit) can execute per unit of time.

Reviewer 2: ADd a link to the CLOPS code here https://github.com/qiskit-community/qiskit-device-benchmarking/tree/main/qiskit_device_benchmarking/clops

> * **CLOPS** (or CLOPS_h) - Circuit layer operations per second, is a measure of how many layers of a 100x100 circuit (hardware-aware circuit) a QPU (quantum processing unit) can execute per unit of time.

Author: ```suggestion
* **CLOPS** (or CLOPS_h) - Circuit layer operations per second, is a measure of how many layers of a 100x100 circuit (hardware-aware circuit) a QPU (quantum processing unit) can execute per unit of time. Find a [code example](https://github.com/qiskit-community/qiskit-device-benchmarking/tree/main/qiskit_device_benchmarking/clops#readme) in the Qiskit Community GitHub.
```

* **Status** - The QPU status.
* **Region** - Location of the data center where your data and experiments will be hosted and processed.
* **QPU version** - The version number of a QPU in the form `major.minor.revision`. See [QPU versioning](#qpu-versioning) for details on how this number is assigned.
* **Processor type** - Reflects the [topology](/docs/guides/qpu-information) and indicates the approximate qubit count.
* **Basis gates** <span id="native-gates-on-platform"></span> - Each [processor family](/docs/guides/processor-types) has a native gate set. By default, the QPUs in each family only support running the gates and operations in the native gate set. Thus, every gate in the circuit must be translated (by the transpiler) to the elements of this set. Note that the non-unitary operations are not listed here; use the [method in Qiskit](/docs/guides/get-qpu-information#native-gates-with-qiskit) to see all native gates and operations for a QPU. See a list of all native gates in [this table](#native-gates).
* **Total pending jobs** - The total number of jobs that you have submitted to this QPU.

> * **Median 2Q error** (Heron: CZ, Eagle: ECR) - [Average gate fidelity](/docs/api/qiskit/quantum_info#average_gate_fidelity) of the two-qubit operation from randomized benchmarking. Measured in "isolation": batches with a minimum separation of two qubits between edges. This randomized benchmarking uses alternating layers of single-qubit Cliffords and two-qubit gates, and thus the final 2Q error value includes the error of the layer of single-qubit Cliffords.

Reviewer 2: Can we add at the end here a link to an example notebook https://github.com/qiskit-community/qiskit-device-benchmarking/blob/main/notebooks/device_rb_w_lf.ipynb

*   Small electronics changes that don’t affect operation.
*   QPU software updates.

## QPU details

The first section of the QPU information card provides the following QPU details: 

*   **Name** - The unique name assigned to a specific QPU. QPUs hosted on IBM Cloud® have names that begin with `ibm_*`. All QPUs are given a city name, e.g., `ibm_kingston`. This name does not indicate where the actual QPU is hosted. They are named after IBM&reg; locations around the world.
*   **Qubits** - The number of physical qubits in a QPU.

> *   **2Q error (best)** - The lowest two-qubit (2Q) error on any edge of the device as measured from isolated 2Q randomized benchmarking.

Reviewer: ```suggestion
*   **2Q error (best)** - The lowest two-qubit (2Q) error on any edge of the device from the same batch of measurements we use to calculate the median.
```

*   **Status** - The QPU status.
*   **Region** - Location of the data center where your data and experiments will be hosted and processed.
* **QPU version** - The version number of a QPU in the form `major.minor.revision`. See [QPU versioning](#qpu-versioning) for details on how this number is assigned.
*   **Processor type** - Reflects the [topology](/docs/guides/qpu-information) and indicates the approximate qubit count.
* **Basis gates** <span id="native-gates-on-platform"></span> - Each [processor family](/docs/guides/processor-types) has a native gate set. By default, the QPUs in each family only support running the gates and operations in the native gate set. Thus, every gate in the circuit must be translated (by the transpiler) to the elements of this set. Note that the non-unitary operations are not listed here; use the [method in Qiskit](/docs/guides/get-qpu-information#native-gates-with-qiskit) to see all native gates and operations for a QPU. See a list of all native gates in [this table](#native-gates).
* **Total pending jobs** - The total number of jobs that you have submitted to this QPU.

> * **Median CZ error** (Heron)

Reviewer 2: see below

> * **Median ECR error** (Eagle)

Reviewer 2: I would have this as `**Median 2Q error** (CZ - Heron, ECR - Eagle) - Fidelity of the 2Q operation measured in "isolation" (in batches with a minimum separation of 2 qubits between edges)` ...

It's Median *of* the average gate fidelity which I know is confusing but what you have is correct. 

Can you also add (wording might need work)

"We use randomized benchmarking to measure the average gate fidelity structured such that there is an alternating layers of 1Q Cliffords and 2Q gates. The final value for the error the 2Q gates includes the error of the this layer of 1Q Cliffords. "

> * **Median ECR error** (Eagle)

Reviewer: ```suggestion
* **Median 2Q error** (CZ - Heron, ECR - Eagle) - [Average gate fidelity](https://quantum.cloud.ibm.com/docs/api/qiskit/quantum_info#average_gate_fidelity) of the 2Q operation from randomized benchmarking. Measured in "isolation": batches with a minimum separation of 2 qubits between edges. This randomized benchmarking uses alternating layers of 1Q Cliffords and 2Q gates, thus the final 2Q error value includes the error of the layer of 1Q Cliffords.
```

> * **Median SX error**

Reviewer: ```suggestion
* **Median SX error** - [Average gate fidelity](https://quantum.cloud.ibm.com/docs/api/qiskit/quantum_info#average_gate_fidelity) of the √X (SX) gate from randomized benchmarking, measured simultaneous on all qubits. Our randomized benchmarking sequence includes SX, ID and X gates, and we assume their errors are the same.

```

> * **Median readout error**

Reviewer 2: "Fidelity of the readout operation. Readout error is measured by preparing the qubit in the 0 (1) state and measuring the probability of an output in the 1 (0) state. The reported value is the average of these two errors."

Median here means median over all qubits

> * **Median readout error**

Reviewer: ```suggestion
* **Median readout error** - Fidelity of the readout operation. Readout error is measured by preparing the qubit in the 0 (1) state and measuring the probability of an output in the 1 (0) state. The reported value is the average of these two errors. Median over all qubits.
```

* **Median T1** (Relaxation Time) - The T1 time represents the average duration a qubit remains in its excited state $|1\rangle$ before decaying to its ground state $|0\rangle$ due to energy relaxation. This parameter is used to characterize the qubit's energy relaxation behavior, and is expressed in units of seconds (s).

> * **Median T2** (Dephasing Time) - The T2 time denotes the timescale over which a qubit maintains phase coherence of a superposition between the $|0\rangle$ and $|1\rangle$ states. It accounts for both energy relaxation and pure dephasing processes, providing insight into the qubit's coherence properties.

Reviewer 2: Add that we report the T2 from a Hahn echo sequence.

Reviewer: ```suggestion
* **Median T2** (Dephasing Time) - The T2 time denotes the timescale over which a qubit maintains phase coherence of a superposition between the $|0\rangle$ and $|1\rangle$ states. It accounts for both energy relaxation and pure dephasing processes, providing insight into the qubit's coherence properties. We report the T2 from a Hahn echo sequence.
```

> * **Median T2** (Dephasing Time) - The T2 time denotes the timescale over which a qubit maintains phase coherence of a superposition between the $|0\rangle$ and $|1\rangle$ states. It accounts for both energy relaxation and pure dephasing processes, providing insight into the qubit's coherence properties.

Reviewer: ```suggestion
* **Median T2** (Dephasing Time) - The T2 time denotes the timescale over which a qubit maintains phase coherence of a superposition between the $|0\rangle$ and $|1\rangle$ states. It accounts for both energy relaxation and pure dephasing processes, providing insight into the qubit's coherence properties. We report the T2 from a Hahn echo sequence.
```

* **Prob meas0 prep1**: This parameter indicates the probability of measuring a qubit in the $|0\rangle$ state when it was intended to be prepared in the $|1\rangle$ state, denoted as $P(0|1)$. It reflects errors in state preparation and measurement (SPAM), particularly measurement errors in superconducting qubits.
* **Prob meas1 prep0**: Similarly, this parameter represents the probability of measuring a qubit in the $|1\rangle$ state when it was intended to be prepared in the $|0\rangle$ state, denoted as $P(1|0)$. Like prob_meas0_prep1, it reflects SPAM errors, with measurement errors being the predominant contributor in superconducting qubits.
* **Readout length (ns)**: The readout_length specifies the duration of the readout operation for a qubit. It measures the time from the initiation of the measurement pulse to the completion of signal digitization, after which the system is ready for the next operation. Understanding this parameter is crucial for optimizing circuit execution, especially when incorporating mid-circuit measurements.

> * **ID error**:

Reviewer: ```suggestion
* **ID error** - Identity gate error, measured from randomized benchmarking. The identify gate is a delay of duration equal to the duration of the SX and X gates. Our randomized benchmarking sequence includes SX, ID and X gates, and we assume their errors are the same.
```

> * **Single-qubit gate length (ns)**:

Reviewer 2: This doesn't make any sense: "measures the time from the initiation of the measurement pulse to the completion of signal digitization, after which the system is ready for the next operation."

"Duration of a single-qubit gate operation." is good enough

> * **Single-qubit gate length (ns)**:

Reviewer: ```suggestion
* **Single-qubit gate length (ns)** - Duration of a single-qubit gate operation.
```

> * **RX error** (Heron):

Reviewer: ```suggestion
* **RX error** - RX gate error, averaged over the RX angles using a variant of randomized benchmarking.
```

> * **Z-axis rotation (rz) error**:

Reviewer: ```suggestion
* **Z-axis rotation (RZ) error** - Error of our virtual RZ gate. Reported as all 0's since these are performed in software.
```

> * **√x (sx) error**:

Reviewer 2: Error

note to see the above definition in the earlier section

> * **√x (sx) error**:

Reviewer: do you mean this?
`Error in our SX gate averaged over the SX angles using a variant of randomized benchmarking for arbitrary unitaries.`

or the "virtual SX" one?

* **Pauli-X error**:
* **Gate length (ns)**:

> * **RZZ error**:

Reviewer 2: Error in our RZZ gate averaged over the RZZ angles using a variant of randomized benchmarking for arbitrary unitaries.

> * **Operational** (Yes/No)

Reviewer: some information is for "single qubit", some is for "two qubits". it is more organized to keep all the one-qubit information together and then the two qubit information. I tried to edit, but it generates branches for me instead. I will maybe submit one of those for you to merge into this branch.

> * **Operational** (Yes/No)

Author: I see. I wonder if you should also ask the UI folks to rearrange the table in a similar fashion, so that the order in which we list them in the docs matches the order on the UI @marcolincs ?

The readout_length specifies the duration of the readout operation for a qubit. It measures the time from the initiation of the measurement pulse to the completion of signal digitization, after which the system is ready for the next operation. Understanding this parameter is crucial for optimizing circuit execution, especially when incorporating mid-circuit measurements.

<span id="finite-duration-discrete"></span>
### ID error / √x (sx) error / Pauli-X error

Error in the finite-duration discrete one-qubit gates, measured from randomized benchmarking. The randomized benchmarking sequence includes SX, ID, and X gates, and it is assumed their errors are the same. The ID gate is a delay of duration equal to the duration of the √X and X gates.

### Single-qubit gate length (ns)

Duration of a single-qubit gate operation.

### RX error

Error in the RX gate, averaged over the RX angles using a variant of randomized benchmarking.

### Z-axis rotation (RZ) error

Error in the virtual RZ gate. Reported as all 0 since these are performed in software.

### Operational

Indicates whether the qubit can be utilized in circuits.

### Gate length (ns)

Duration of the two-qubit gate operation.

> ### 2Q error (best) (Heron: CZ, Eagle: ECR)

Author: ```suggestion
### 2Q error (Heron: CZ, Eagle: ECR)
```

### CLOPS (or CLOPS_h) 

Circuit layer operations per second, is a measure of how many layers of a 100x100 circuit (hardware-aware circuit) a QPU (quantum processing unit) can execute per unit of time. Find the [CLOPS code](https://github.com/qiskit-community/qiskit-device-benchmarking/tree/main/qiskit_device_benchmarking/clops#readme) in the Qiskit Community GitHub.

### Status

The QPU status; for example, `Online`, `Paused`, `Offline`, and so on.

### Region

Location of the data center where your data and experiments will be hosted and processed.

### QPU version

The version number of a QPU in the form `major.minor.revision`. See [QPU versioning](#qpu-versioning) for details on how this number is assigned.

### Processor type

Reflects the [topology](/docs/guides/qpu-information) and indicates the approximate qubit count.

> ### Basis gates<span id="native-gates-on-platform"></span>

Reviewer 3: Ditto here and in all the headings

Author: Done in [this commit](https://github.com/Qiskit/documentation/pull/3803/commits/bebbf941e5d2462eb1eb69c554027c957927b88e)

You can customize the data that is shown in each view, using the drop-down menus. For example, in the map view, you can choose the data you want to see for qubits and connections. The colored bars associated with the diagram or graph indicate the range that is shown, with the average value marked. The color maximum and minimum change depending on the QPU.

To download calibration data as a CSV file, click the download icon in the upper right corner of the Calibration data section.

In addition to the information provided in the Details section of the card, the Calibration data section also includes the following:

[Topology diagram or coupling map](#coupling-map) | [Readout assignment error](#readout-assignment-error) | [Prob meas0 prep1](#prob-meas0-prep1) | [Prob meas1 prep0](#prob-meas1-prep0) | [Readout length (ns)](#readout-length-ns) | [ID error / √x (sx) error / Pauli-X error](#finite-duration-discrete) | [Single-qubit gate length (ns)](#single-qubit-gate-length-ns) | [RX error](#rx-error) | [Z-axis rotation (RZ) error](#z-axis-rotation-rz-error) | [Operational](#operational) | [Gate length (ns)](#gate-length-ns) | [2Q error](#2q-error) | [RZZ error](#rzz-error-heron)

> ### <span id="coupling-map">Topology diagram or coupling map</span>

Reviewer 3: To make sure this doesn't create any problems with translations, could we have the `span` in a dedicated line immediately before the heading? 🙏. Like in other headings from this same page (e.g. `2Q error (Heron: CZ, Eagle: ECR)` from this same file)

<span id="available"></span>
## View your resources

To find your available QPUs, open the [Compute resources](https://quantum.cloud.ibm.com/computers) page (make sure you are signed in).  Note that your selected region might impact the QPUs listed. Click a QPU to view its details.

You can also view your available QPUs by using the [backends API.](/docs/api/qiskit-ibm-runtime/qiskit-runtime-service#backends) For example, the following code will return all of the backends that the specified instance (`my_instance`) can access:

```python
   QiskitRuntimeService(instance="my_instance_CRN")
   service.backend()
```

<span id="native-gates"></span>
## Table of native gates and operations

| Operation category |  Name |
| :--- | :--- |

> | Single-qubit gates | [RZ](/docs/api/qiskit/qiskit.circuit.library.RZGate), [SX](/docs/api/qiskit/qiskit.circuit.library.SXGate), [X](/docs/api/qiskit/qiskit.circuit.library.XGate), [ID](/docs/api/qiskit/qiskit.circuit.library.IGate), [delay](/docs/api/qiskit/circuit#qiskit.circuit.Delay) |

Reviewer 3: What do you think about using code for the operation names inside the links? For example, I think the Globalization Pipeline will translate names like `delay`, `measure`, `reset` otherwise.

Example:
```md
[`delay`](/docs/api/qiskit/circuit#qiskit.circuit.Delay)
```
